File: Dialogs/DialogDemo/app/src/main/java/edu/cs4730/dialogdemo/myAlertDialogFragment.java
package edu.cs4730.dialogdemo;

import android.app.Activity;
import android.content.Context;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

/*
 * this extends a Dialog fragment to create an alterdialog fragment
 */


public class myAlertDialogFragment extends DialogFragment {

    private OnDialogFragmentListener mListener;


    public static myAlertDialogFragment newInstance(int title) {
        myAlertDialogFragment frag = new myAlertDialogFragment();
        Bundle args = new Bundle();
        args.putInt("title", title);
        frag.setArguments(args);
        return frag;
    }
    
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        int title = getArguments().getInt("title");

        return new AlertDialog.Builder(getActivity())
                //.setIcon(R.drawable.alert_dialog_icon)
                .setTitle(title)
                .setPositiveButton(R.string.alert_dialog_ok,
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog, int whichButton) {
                            mListener.doPositiveClick();
                        }
                    }
                )
                .setNegativeButton(R.string.alert_dialog_cancel,
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog, int whichButton) {
                            mListener.doNegativeClick();
                        }
                    }
                )
                .create();
    }

    //all the callback stuff.
    public interface OnDialogFragmentListener {
        void doPositiveClick();
        void doNegativeClick();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        Activity activity = getActivity();
        try {
            mListener = (OnDialogFragmentListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }
    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }



}

File: Dialogs/DialogDemo_kt/app/src/main/java/edu/cs4730/dialogdemo_kt/myAlertDialogFragment.kt
package edu.cs4730.dialogdemo_kt

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import java.lang.ClassCastException

/**
 * this extends a Dialog fragment to create an alterdialog fragment
 */
class myAlertDialogFragment : DialogFragment() {
    private var mListener: OnDialogFragmentListener? = null
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = requireArguments().getInt("title")
        return AlertDialog.Builder(requireActivity()) //.setIcon(R.drawable.alert_dialog_icon)
            .setTitle(title)
            .setPositiveButton(
                R.string.alert_dialog_ok
            ) { dialog, whichButton -> mListener!!.doPositiveClick() }
            .setNegativeButton(
                R.string.alert_dialog_cancel
            ) { dialog, whichButton -> mListener!!.doNegativeClick() }
            .create()
    }

    //all the callback stuff.
    interface OnDialogFragmentListener {
        fun doPositiveClick()
        fun doNegativeClick()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity: Activity? = activity
        mListener = try {
            activity as OnDialogFragmentListener?
        } catch (e: ClassCastException) {
            throw ClassCastException(
                activity.toString()
                        + " must implement OnFragmentInteractionListener"
            )
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    companion object {
        fun newInstance(title: Int): myAlertDialogFragment {
            val frag = myAlertDialogFragment()
            val args = Bundle()
            args.putInt("title", title)
            frag.arguments = args
            return frag
        }
    }
}
